compiler/rustc_resolve/src/late.rs

```diff
@@ -4686,10 +4759,14 @@ impl<'a, 'ast, 'ra, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
         // Move back up.
         self.parent_scope.module = orig_module;
         for _ in 0..num_macro_definition_ribs {
-            self.ribs[ValueNS].pop();
+            // pop `MacroDefinition`
             self.label_ribs.pop();
         }
-        self.last_block_rib = self.ribs[ValueNS].pop();
+        let block_rib = self.ribs[ValueNS].pop(); // pop `RibKind::Block`
+        if !is_module_block {
```

Comment:
`Module`s are indeed not created for some blocks, but that's only a performance optimization.
There should be zero semantic difference between module blocks and non-module blocks, but this PR seems to add some.

Reply:
> but this PR seems to add some

They're only to generate `last_block_rib` which is only used for diagnostics, specifically to display information about non-module blocks:

https://github.com/rust-lang/rust/blob/6bcdcc73bd11568fd85f5a38b58e1eda054ad1cd/compiler/rustc_resolve/src/late/diagnostics.rs#L853-L870